lira/bundle_inputs.py

```
@@ -0,0 +1,71 @@
+                                                                taxon_id=taxon_id,
+                                                                read_1=r1_hashes,
+                                                                read_2=r2_hashes), encoding='utf8')
+
+
+def get_optimus_workflow_input_hash(bundle):
+    """ Return a string that is a concatenation of bundle-specific Optimus workflow inputs
+    where the file hashes are for each set of read_1, read_2 and index_1 files in the order
+    of their flow cell lane number. For example:
+    {sample_id}{taxon_id}{lane1_r1}{lane1_r2}{lane1_i1}{lane2_r1}{lane2_r2}{lane2_i1}
+
+    Each "file hash" is a concatenation of {sha1}{sha256}{s3_etag}{crc32c}
+    """
+    sample_id = metadata_utils.get_sample_id(bundle)
+    taxon_id = metadata_utils.get_sample_id(bundle)
+    inputs = '{sample_id}{taxon_id}'.format(sample_id=sample_id, taxon_id=taxon_id)
+    fastq_files = [f for f in bundle.files.values() if str(f.format).lower() in ('fastq.gz', 'fastq')]
+    fastq_dict = tenx_utils.create_fastq_dict(fastq_files)
+    sorted_lanes = sorted(fastq_dict.keys(), key=int)
+    for lane in sorted_lanes:
+        r1_hashes = get_hashes_from_file_manifest(fastq_dict[lane]['read1'])
+        r2_hashes = get_hashes_from_file_manifest(fastq_dict[lane]['read2'])
+        file_hashes = '{0}{1}'.format(r1_hashes, r2_hashes)
+        if fastq_dict[lane].get('index1'):
+            i1_hashes = get_hashes_from_file_manifest(fastq_dict[lane]['index1'])
+            file_hashes += i1_hashes
+        inputs += file_hashes
+    return bytes(inputs, encoding='utf8')
+
+
+WORKFLOW_INPUTS = {
+    'AdapterSmartSeq2SingleCell': get_smartseq2_workflow_input_hash,
+    'AdapterOptimus': get_optimus_workflow_input_hash
+}
+
+
+def create_workflow_inputs_hash_label(workflow_name, bundle_id, bundle_version, dss_url):
+    get_inputs_fn = WORKFLOW_INPUTS.get(workflow_name, None)
+    if get_inputs_fn:
```

This means the workflows that are not in the "supported mapping" `WORKFLOW_INPUTS` will be ignored and won't get a hash. I think we need to discuss how we wanna handle that case. Do we really want to start the workflows that doesn't have a deterministic set of inputs via Lira? If so, what are the use cases? If not, can we just raise an error here?